### PR TITLE
Fix issue with classifier thinking something was a partially written generic type.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/13174

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Workspaces/CSharp/Portable/Classification/Classifiers/SyntaxTokenClassifier.cs [hunk 1]
```diff
@@ -33,15 +34,56 @@ public override IEnumerable<int> SyntaxTokenKinds
...
+            if (identifierName.IsParentKind(SyntaxKind.MemberBindingExpression))
+            {
+                // anything?.Identifier is never a generic type.
+                return false;
+            }
+
+            if (identifierName.IsMemberAccessExpressionName())
+            {
+                // ?.X.Identifier   or  ?.X.Y.Identifier  is never a generic type.
+                if (identifier.Parent.IsParentKind(SyntaxKind.ConditionalAccessExpression))
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/Classification/Classifiers/SyntaxTokenClassifier.cs]: Does this need to walk up a whole chain?

**Comment by Pilchie** [comment]
:+1:

File: src/Workspaces/CSharp/Portable/Classification/Classifiers/SyntaxTokenClassifier.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/Classifiers/SyntaxTokenClassifier.cs]: No, and i'll add a comment explaining.  When you have IsInPartiallyWrittenGeneric, you get the identifier token immediately proceeding the ```<```.  In the ```?.``` case, the identifier will either be parented by the member binding, or will be the right name of the member-access that parented by the conditional expression.  You're never on the 'left' of these recursive structures, so you don't need ot keep walking up.

**Pull request merged**